### ISSUE TITLE
[Question] How to use this with selenium-python on my current page?

### ISSUE THREAD

**Issue by M-Zubair10** [QIN2DIM/hcaptcha-challenger#23]
How to use this with selenium-python on my current page?

**Comment by QIN2DIM** [comment]
Imitate and modify.
https://github.com/QIN2DIM/hcaptcha-challenger/blob/7fc13d9e193ea8cb527af91d3bbf1737d1ac10c4/src/apis/scaffold/challenge.py#L18-L75

**Comment by M-Zubair10** [comment]
Can you make standalone class like class.solve() to solve captcha on
current page

**Comment by QIN2DIM** [comment]
@M-Zubair10 in process🐒

**Comment by M-Zubair10** [comment]
Thanks a lot.

**Comment by Revadike** [comment]
maybe u can also make a standalone proxy server exe/linux

**Comment by M-Zubair10** [comment]
Nah!
I only need to use it during my webscraping in selenium
So needed a pypi release with function solve to automatically solve captcha
on current selenium page

I actually done this with yolov3 using image ai but the result ratio is 2
out of 5 which is bad, your model is quite good but i can't load it in
imageai since i don't have a knowledge of tensorflow, keras for now.

**Comment by QIN2DIM** [comment]
@M-Zubair10 The solution for this project uses the ONNX compression model, and you can call it directly using the python-opencv interface without caring about the details of deep learning. If you look at requirement.txt, you will see that torch or tensorflow is missing.

@M-Zubair10  @Revadike  btw I don't have a good grasp of pypi and container server, which will take some time to learn. and then, I've been busy with school lately, so I may not be able to push it anytime soon.🤦‍♂️

**Comment by QIN2DIM** [comment]
After a round of research I found that you guys are actually asking the same question...

**Comment by QIN2DIM** [comment]
这个项目名叫 `hcaptcha-challenger` 而非 `hcaptcha-solver`。我的本意是搭建一个服务接口用以演示 `YOLOv5(ONNX) 嵌入式解决方案` 带来的提升，我只在乎识别的速度以及挑战的通过效率。

对于 @Revadike 这位老哥的问题，用什么方法对抗 hCaptcha 并不重要，重要的是获取挑战结束后返回的 TOKEN 。

对于 @M-Zubair10 的问题，如果你只是想在你编写的 selenium 程序里使用这个挑战方法，你可以按照我上面说的，根据 demo 改一下上下文业务既可。如果你要把它打包成 pypi，编码难度极高。

**Comment by M-Zubair10** [comment]
Can you again send this in english?

**Comment by QIN2DIM** [comment]
ah~ @Revadike  I know what you mean, you need a cross programming language solution.

**Comment by QIN2DIM** [comment]
@M-Zubair10 ah - - I think I didn't express my meaning accurately. 

The coding difficulty I mentioned yesterday is mainly due to the fact that `hcaptcha-challenger` cannot be started in various selenium contexts. Because the challenge label is multilingual, when you use selenium on a PC to trigger the challenge, the `lang of label` depends on the `--lang` parameter of the `WebDriver Options`, and when you use `selenium` on linux, the `lang of label` depends on the value of the process environment variable `LANGUAGE`.

Obviously all such recognition operations involve label matching, and if the label text is translated into other languages because of multilingual issues, not only will the model fail, but the whole challenge logic will be seriously flawed.

I would like to make this solution available to developers using a variety of languages with a single set of code, however, the two factors I just mentioned that determine the `lang of label` are set before the `WebDriver` starts. To reference `hcaptcha-challenger`'s method in the process for a challenge, you would have to do redundant transcoding, which is unnecessary.

**Comment by M-Zubair10** [comment]
Thank you for the clarification.

**Comment by izoomrud** [comment]
> For @Revadike，it doesn't matter what method is used against hCaptcha, it's important to get the TOKEN returned at the end of the challenge.

How can I get TOKEN from there?

I tried putting `token = ctx.find_element_by_tag_name('iframe').get_attribute("data-hcaptcha-response")` after `self.log("Challenge Success")` in core.py but it doesn't work.t work

**Comment by QIN2DIM** [comment]
>  but it doesn't work.
@izoomrud sure

**Comment by izoomrud** [comment]
> > but it doesn't work.
> 
> @izoomrud sure

ctx.switch_to.default_content()
        time.sleep(3)
        from pprint import pprint
        token = ctx.find_elements(By.XPATH, "//iframe[@title='widget containing checkbox for hCaptcha security challenge']")
        for value in token:
            file = open('token.txt', 'a')
            file.write(value.get_attribute('data-hcaptcha-response'))

.|.

**Comment by izoomrud** [comment]
> > but it doesn't work.
> 
> @izoomrud sure

```
ctx.switch_to.default_content()
        time.sleep(3)
        token = ctx.find_elements(By.XPATH, "//iframe[@title='widget containing checkbox for hCaptcha security challenge']")
        for value in token:
            file = open('token.txt', 'a')
            file.write(value.get_attribute('data-hcaptcha-response'))
```

.|.

**Comment by QIN2DIM** [comment]
@izoomrud 有被可爱到，路漫漫其修远兮哈哈哈哈

**Comment by LanLan69** [comment]
execute javascript to get the token. 
just use getAttribute("data-hcaptcha-response") and then set the token to the webpage's title and do webdriver.title 
and that will work

**Comment by shahzain345** [comment]
Well, what u can do is execute javascript in your webdriver. You can use the `hcaptcha.getResponse()` method to get the response token, which you can then use in your script. The code should look like this.
```py
token = ctx.execute_script("return hcaptcha.getResponse();")
```

**Comment by yeshenshuijiao** [comment]
可以使用playwright吗
